libecole/include/ecole/scip/model.hpp

```diff
@@ -133,77 +151,108 @@ class Model {
 
 namespace internal {
 
-// SFINAE default class for no available cast
-template <typename To, typename From, typename = void> struct Caster {
-	static To cast(From) { throw Exception("Cannot convert to the desired type"); }
-};
-
-// SFINAE class for available cast
-template <typename To, typename From>
-struct Caster<To, From, std::enable_if_t<std::is_convertible<From, To>::value>> {
-	static To cast(From val) { return static_cast<To>(val); }
-};
-
-// Pointers must not convert to bools
-template <typename From> struct Caster<bool, std::remove_cv<From>*> {
-	static bool cast(From) { throw Exception("Cannot convert pointers to bool"); }
-};
-
-// Convert charachter to string
-template <> std::string Caster<std::string, char>::cast(char);
-
-// Convert string to character
-template <> char Caster<char, char const*>::cast(char const*);
-template <> char Caster<char, std::string>::cast(std::string);
+template <class To, class From>
+typename std::enable_if<std::is_same<To, From>::value == true, To>::type
+narrow_cast(From v) {
+	return v;
+}
```

Comment:
Well, we can let it happen, although that will trigger two unnecessary casts. It is more an optimization concern. They impact may be minor though, so maybe that's not necessary. However I do not think the cost for readability is high, it is quite self-explicit what is happening here.

Reply:
In my experiment, very small optimization was getting rid of it, but there is a warning `useless-cast` enabled and it may not like it anyways.

Note that if this was a struct, you can partial specialize with `template<typename T> Caster<T, T> {...}` (does not work with functions I believe).
Might be useful with the `Caster` (see my previous comment).